### ISSUE TITLE
Function app dependencies need to be downloaded to a common location for local developement

### ISSUE THREAD

**Issue by Francisco-Gamino** [Azure/azure-functions-powershell-worker#203]
Currently, for the local scenario, we are downloading the function app to LocalApplicationData\AzureFunctions\FunctionAppName\ManagedDependencies. However, we should redesign this to behave more like a Nuget cache. 

The proposed location could be:
LocalApplicationData\AzureFunctions\<Feature>\<Language>, e.g.,
LocalApplicationData\AzureFunctions\ManagedDependencies\PowerShell

I

**Comment by joeyaiello** [comment]
Quoting @TylerLeonhardt, @SteveL-MSFT, and @daxian-dbw, this is infeasible as it is essentially equivalent to creating a GAC.

If you want to talk in more detail on it, find one of them offline.

**Comment by Francisco-Gamino** [comment]
@joeyaiello : Yes, we should talk more. This feature/enhancement was requested by @pragnagopa.

In the current implementation, we do multiple downloads (one per function app). I think we can do a better. I will reopen the issue so we can discuss it. 

cc: @TylerLeonhardt @SteveL-MSFT @daxian-dbw @anirudhgarg @eamonoreilly

**Comment by AnatoliB** [comment]
If you want to go ahead and discuss this further, please involve me in this conversation. I do have some concerns (probably similar to "equivalent to creating a GAC").

**Comment by pragnagopa** [comment]
Here is my perspective:

This is an improvement work item. Size of ManagedDependencies folder for a single function app is about ~120MB.  If there are more dependencies or user is trying different function app this adds up over a period of time as we do not clean up these folders.  Also, when the ManagedDependencies feature  expands to other languages like python, javascript etc CLI will be downloading same modules over and over.

When running in Azure, to improve startup time we need to move to a model similar to ExtensionBundles instead of downloading dependencies to user storage. I understand that would be a future work time but addressing this issue will put us in the right direction for when we do switch to a model where dependencies are completely managed by us

@Francisco-Gamino / @eamonoreilly  / @anirudhgarg - Please feel free to move this to backlog if you think it is not a higher priority item. We do have to revisit the design when adding ManagedDependecies feature to other languages.

**Comment by eamonoreilly** [comment]
I think for now we should add a warning if the size of LocalApplicationData\AzureFunctions is > 1GB when a new PowerShell function app is created (func init). The warning could say the following:

" The size of <path>\AzureFunctions is currently greater than 1GB for managed dependency downloads. Please delete any unused function app folders from this location to recover space."
